packages/web/pages/player/[username].tsx

```diff
@@ -180,12 +180,17 @@ export const getStaticPaths: GetStaticPaths<QueryParams> = async () => {
 export const getStaticProps = async (
   context: GetStaticPropsContext<QueryParams>,
 ) => {
-  const username = context.params?.username;
-  const player = await getPlayer(username);
-
+  const username = context.params?.username?.toLowerCase();
```

Comment:
Maybe it would be great to redirect to the page with the username correctly lowercase, so that if the user share the URL it is the good one.

https://maxschmitt.me/posts/next-js-redirects/

`if (context.params?.username !== context.params?.username?.toLowerCase();) `

Reply:
> The regex check is only present on the frontend, so if we manually call the graphql endpoint with a bad formated username, it will be accepted.
> 
> The check should be done on the backend side. I'm not sure hasura can natively do that, so maybe we should have an handler for that.

Yup, agreed; I spent about an hour playing around with regex / custom checks in postgres / hasura before throwing up my hands. I have another idea that should be cleaner.

Also agree about the redirect; it seems we should be somehow communicating to the user what the correct link is.

Reply:
I implemented both changes; have a look @pakokrew